parmed/amber/_amberparm.py

```diff
@@ -2286,6 +2286,62 @@ def box(self, value):
         else:
             self._box = np.array(value).reshape((-1, 6))[0]
 
+    @property
+    def _ucell(self):
+        """Return the unit cell vectors as a packed 3x3 matrix.
+        
+        If no box is set, return None.
+        """
+        if not self.hasbox:
+            return None
+        ucell = np.zeros(9, dtype=self.box.dtype)
+        # Because we're not vectorizing anything we'll do math with Python
+        # builtin types and then return as a NumPy array.
+        #
+        # Math is taken from cpptraj/src/Box.cpp Box::CalcUcellFromXyzAbg
+        #
+        deg2rad = math.pi / 180.0
+        x, y, z, alp, bet, gam = map(float, self.box)
+        ucell[0] = x
+        if isclose(gam, 90.0):
+            ucell[4] = y
+        else:
+            ucell[3] = y * cos(gam * deg2rad)
+            ucell[4] = y * sin(gam * deg2rad)
+        if not isclose(bet, 90.0):
+            ucell[6] = z * cos(alp * deg2rad)
+        yz_cos_alp = 0.0
+        if not isclose(alp, 90.0):
+            yz_cos_alp = x * y * cos(alp * deg2rad)
+        ucell[7] = (yz_cos_alp - ucell[6] * ucell[3]) / ucell[4]
+        ucell[8] = sqrt(z**2 + ucell[6]**2 + ucell[7]**2)
+        return ucell
```

Comment:
Maybe you can re-use some of the functions in `geometry.py`?

Reply:
Indeed I can.

The next item for discussion - For now I elected to return a float in A^3 rather than a Quantity. Is this objectionable? IMO the Quantity class is useful for internal calculations but I find it cumbersome to require an additional import just to interpret an attribute or method result.

Reply:
Alternative -- Change `volume` attribute to `volume(use_units)` method and let the user decide.

Reply:
I've been trying to offer quantitied properties by prefixing with `u`, so `volume` would be a property returning a float as cubic angstroms while `uvolume` would return `volume * u.angstroms**3`.

Reply:
Done.